File: skills/ai-attack-simulator.js
/**
 * skills/ai-attack-simulator.js — AI-Driven Attack Simulator
 * Wrapper Node.js para os scripts Python de simulação de ataques
 * Hardened: sanitize de input para prevenir command injection
 */

const { execSync } = require('child_process');
const path = require('path');

const SCRIPTS_DIR = path.join(__dirname, '..', 'scripts');

// Sanitiza input: remove caracteres perigosos do shell
function sanitize(input) {
  if (typeof input !== 'string') return '';
  // Remove caracteres que podem ser usados para injection
  return input.replace(/[;&|`$(){}[\]\\'"<>!]/g, '').trim();
}

function simulateAttack(target, scenario, parameters = null) {
  const safeTarget = sanitize(target);
  const safeScenario = sanitize(scenario);
  if (!safeTarget || !safeScenario) {
    throw new Error('Target ou scenario inválido (caracteres perigosos detectados)');
  }
  
  const paramsStr = parameters ? JSON.stringify(parameters) : '{}';
  // Usa array de args em vez de string para evitar shell injection
  const cmd = [
    'python3',
    path.join(SCRIPTS_DIR, 'ai_attack_simulator.py'),
    '--action', 'simulate',
    '--target', safeTarget,
    '--scenario', safeScenario,
    '--parameters', paramsStr
  ];
  
  const result = execSync(cmd.join(' '), { encoding: 'utf8', timeout: 60000 });
  return JSON.parse(result);
}

function monitorAttack(attackId) {
  const safeId = sanitize(attackId);
  if (!safeId) throw new Error('Attack ID inválido');
  
  const cmd = [
    'python3',
    path.join(SCRIPTS_DIR, 'ai_attack_simulator.py'),
    '--action', 'monitor',
    '--target', safeId
  ];
  
  const result = execSync(cmd.join(' '), { encoding: 'utf8', timeout: 30000 });
  return JSON.parse(result);
}

function analyzeResults(attackData) {
  const inputStr = JSON.stringify(attackData);
  const cmd = [
    'python3',
    path.join(SCRIPTS_DIR, 'ai_attack_simulator.py'),
    '--action', 'analyze',
    '--target', inputStr
  ];
  
  const result = execSync(cmd.join(' '), { encoding: 'utf8', timeout: 30000 });
  return JSON.parse(result);
}

module.exports = { simulateAttack, monitorAttack, analyzeResults };

File: scripts/ai_attack_simulator.py
import json
import subprocess
import argparse
import sys
from datetime import datetime

# Importa analysis_engine diretamente para evitar subprocess com JSON
sys.path.insert(0, 'scripts')
from analysis_engine import analyze_attack

class AIAttackSimulator:
    def __init__(self):
        self.attack_id = None
        self.start_time = None
        
    def simulate_attack(self, target, scenario, parameters=None):
        self.attack_id = f"attack_{datetime.now().strftime('%Y%m%d%H%M%S')}"
        self.start_time = datetime.now()
        
        cmd = ["python3", "scripts/attack_engine.py", "--target", target, "--scenario", scenario]
        if parameters:
            for k, v in parameters.items():
                cmd.extend(["--" + k, str(v)])
        
        try:
            result = subprocess.run(cmd, capture_output=True, text=True)
            return json.loads(result.stdout)
        except Exception as e:
            return {"error": str(e)}
            
    def monitor_attack(self, attack_id):
        cmd = ["python3", "scripts/monitor_engine.py", "--attack-id", attack_id]
        result = subprocess.run(cmd, capture_output=True, text=True)
        return json.loads(result.stdout)
        
    def analyze_results(self, attack_data):
        # Chama diretamente em vez de subprocess
        return analyze_attack(attack_data)

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("--action", choices=["simulate", "monitor", "analyze"])
    parser.add_argument("--target")
    parser.add_argument("--scenario")
    parser.add_argument("--parameters", type=json.loads, default="{}")
    args = parser.parse_args()
    
    simulator = AIAttackSimulator()
    
    if args.action == "simulate":
        print(json.dumps(simulator.simulate_attack(args.target, args.scenario, args.parameters)))
    elif args.action == "monitor":
        print(json.dumps(simulator.monitor_attack(args.target)))
    elif args.action == "analyze":
        # Lê JSON do stdin
        input_data = sys.stdin.read()
        try:
            attack_data = json.loads(input_data) if input_data else {"orchestrator": {"start_time": datetime.now().isoformat()}}
        except:
            attack_data = {"orchestrator": {"start_time": datetime.now().isoformat()}}
        print(json.dumps(simulator.analyze_results(attack_data)))
